nginx-controller/nginx/configurator.go

```diff
@@ -171,7 +176,18 @@ func (cnf *Configurator) generateNginxCfg(ingEx *IngressEx, pems map[string]stri
 		servers = append(servers, server)
 	}
 
-	return IngressNginxConfig{Upstreams: upstreamMapToSlice(upstreams), Servers: servers}
+	result := []IngressNginxConfig{}
```

Comment:
We can eliminate this steps. We generate Servers that include upstreams and then return those Servers